### PR TITLE
[BugFix] Fix fold str_to_date with %f failed

### PR DESCRIPTION
**Pull request by Seaven**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 7 / 7 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/util/DateUtils.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/scalar/ConstantOperator.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rewrite/ScalarOperatorFunctions.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 7 / 7 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/util/DateUtils.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/scalar/ConstantOperator.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rewrite/ScalarOperatorFunctions.java|2|2|100.00%|[]|

**Review by liuyehcf — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

File: fe/fe-core/src/test/java/com/starrocks/sql/optimizer/rewrite/ScalarOperatorFunctionsTest.java [hunk 1]
```diff
@@ -315,6 +315,19 @@ public void dateParse() {
         assertEquals("2013-05-17T12:35:10", ScalarOperatorFunctions
                 .dateParse(ConstantOperator.createVarchar("2013-05-17 12:35:10"),
                         ConstantOperator.createVarchar("%Y-%m-%d %H:%i:%s")).getDatetime().toString());
+
+        assertEquals("2013-05-17T12:35:10.000123", ScalarOperatorFunctions
```
Comment (jaogoy) [inline fe/fe-core/src/test/java/com/starrocks/sql/optimizer/rewrite/ScalarOperatorFunctionsTest.java:319]: Is this correct?

**Pull request merged**